app/models/topic.rb

```diff
@@ -1234,6 +1247,9 @@ def muted?(user)
   def self.ensure_consistency!
     # unpin topics that might have been missed
     Topic.where("pinned_until < now()").update_all(pinned_at: nil, pinned_globally: false, pinned_until: nil)
+    Topic.where('bannered_until <= ?', Time.now).find_each do |topic|
+      topic.remove_banner!(Discourse.system_user)
+    end
```

Comment:
@SamSaffron Are we still OK with this pattern or do you prefer having a recurring job that runs regularly like how we do for bookmarks?

Reply:
to me both pinned until and bannered until are fancy topic timers. I would be delighted if these two edge cases just fit into the topic timer system instead of living in a special codepath. ... @martin-brennan any thoughts here.

Reply:
I agree with this...in theory. However topic timers still suffer from the fact that you can only have one at a time, our backend and our UI is built around this. For example we move bannered_until into a topic timer and then you want an auto-close or a temporarily close on the same topic. You can't do it. I think we need to do this in future work, an overhaul of the timer system to allow for multiple timers.

Until then this functionality, while not perfect, is probably the best way to go (copying how pinned until works).